client/src/hooks/useImgUpload.ts

```diff
@@ -0,0 +1,55 @@
+import { useState } from "react";
+import { useMutation } from "@apollo/client";
+
+import { CREATE_S3_SIGNED_URL } from "../graphql/create_s3_signed_url.mutation";
+import { parseError } from "../graphql/client";
+
+/**
+ * Custom hook to handle S3 Upload and generate 'signedURL' and 'objectURL'.
+ * TODO: Handle multiple file uploads.
+ * Should we extend this hook to handle model mutations?
+ */
+export const useImgUpload = () => {
+  const [error, setError] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  const [createS3SignedUrl] = useMutation<{
+    createS3SignedUrl: { s3SignedUrl: string; url: string };
+  }>(CREATE_S3_SIGNED_URL);
+
+  const generateUploadUrls = async (file: File, fileName: string) => {
```

Comment:
This is the core part of s3 url upload. 
Flow: 

1) Call `createS3SignedUrl` mutation which will return the `signedurl` to be used to upload the file + expected url where file can be accessed.
2) upload the file on client side.
3) once upload is complete, return the `expected url` where it will be stored in db.